ui/src/main/java/edu/wpi/grip/ui/codeGeneration/Exporter.java

```diff
@@ -0,0 +1,143 @@
+      out.append(type + "\n");
+    }
+    out.append(step.getOperationDescription().summary() + "\n");
+    return out.toString();
+  }
+
+  public static String createClass(String type, String name, String value) {
+    if (type.contains("Enum")) {
+      return "int " + name + " = Imgproc." + value;
+    } else if (type.equals("Type")) {
+      return "String " + name + " = \"" + value + "\"";
+    } else if (type.equals("List")) {
+      StringBuilder out = new StringBuilder();
+      out.append("double min" + name + " = " + value.substring(1, value.indexOf(",")) + "\n");
+      out.append("double max" + name + " = " + value.substring(value.indexOf(",") + 1, value.lastIndexOf
+          ("]")));
+      return out.toString();
+    } else {
+      return type + " " + name + " = " + value;
+    }
+
+  }
+
+
+  public String getOutputNames(Step step) {
+    StringBuilder out = new StringBuilder();
+    for (OutputSocket output : step.getOutputSockets()) {
+      out.append("    " + output.getSocketHint().getType().getSimpleName() + ": " + output.getSocketHint()
+          .getIdentifier() +
+          " \n");
+    }
+    return out.toString();
+  }
+
+  public void export(Pipeline pipeline, Language lang, File dir, boolean loadLib) {
+    TPipeline tPipeline = new TPipeline(pipeline);
+    TemplateMethods tempMeth = new TemplateMethods();
+    VelocityContext context = new VelocityContext();
+    context.put("pipeline", tPipeline);
+    context.put("tMeth", tempMeth);
+    context.put("fileName", dir.getName().substring(0, dir.getName().lastIndexOf(".")));
+    context.put("loadLib", loadLib);
+    StringBuilder templateDirBuilder = new StringBuilder();
+    templateDirBuilder.append("src/main/resources/edu/wpi/grip/ui/templates/");
+    switch (lang) {
+      case JAVA:
+        templateDirBuilder.append("java");
+        break;
+      case PYTHON:
+        templateDirBuilder.append("python");
+        break;
+      case CPP:
+        templateDirBuilder.append("cpp");
+        break;
+      default:
+        throw new IllegalArgumentException(lang.toString() + " is not a supported language for code generation.");
+    }
+    templateDirBuilder.append("/");
+    final String templateDir = templateDirBuilder.toString();
+    VelocityEngine ve = new VelocityEngine();
+    Properties props = new Properties();
+    props.put("velocimacro.library", templateDir + "macros.vm");
+    ve.init(props);
+    try {
+      switch (lang) {
+        case CPP:
+
+          break;
+        case JAVA:
+          exportJava(ve, templateDir, dir, context);
+          break;
+        case PYTHON:
+
+          break;
+      }
+    } catch (ResourceNotFoundException e) {
+      String error = e.getMessage();
+      String missingOperation = error.substring(error.lastIndexOf("/") + 1, error.lastIndexOf("."));
+      throw new UnsupportedOperationException("The operation " + missingOperation + " is not supported for export to " + lang.toString());
+    }
```

Comment:
It would be really cool if we could make our interface to the templates language-agnostic - i.e. provide the same data structures to templates across all language types. That would allow this code to be the same for every language.

It looks like this code is not far from that, it would benefit from some cleaning to get it all the way there.